mod-inventory/src/main/resources/domain/mod-inventory/features/inventoryFeatureTest.feature

```diff
@@ -50,3 +50,433 @@ Feature: inventory sample
       """
       When method POST
       Then status 201
+
+    Scenario: duplicate Items creation
+
+#     Instance
+      Given path 'inventory/instances'
+      And request
+      """
+      {
+        "source":"FOLIO",
+        "title":"TestInstance",
+        "instanceTypeId":"6312d172-f0cf-40f6-b27d-9fa8feaf332f"
+      }
+      """
+      When method POST
+      Then status 201
+      * def instancelocation = responseHeaders['Location'][0]
+      * def instanceId = instancelocation.substring(instancelocation.lastIndexOf('/') + 1)
+
+#     Holdings
+      Given path 'holdings-storage/holdings'
+      And request
+      """
+      {
+        "instanceId":"#(instanceId)",
+        "permanentLocationId":"184aae84-a5bf-4c6a-85ba-4a7c73026cd5"
+      }
+      """
+      When method POST
+      Then status 201
+      * def holdingsId = response.id
+      * def hrId = response.hrid
+
+#     First Item
+      Given path 'inventory/items'
+      And request
+      """
+      {
+        "status":{"name":"Available"},
+        "holdingsRecordId":"#(holdingsId)",
+        "materialType":{"id":"d9acad2f-2aac-4b48-9097-e6ab85906b25"},
+        "permanentLoanType":{"id":"2e48e713-17f3-4c13-a9f8-23845bb210a4"}
+      }
+      """
+      When method POST
+      Then status 201
+      * def itemId = response.id
+      * def itemHrId = response.hrid
+#     Duplicate Item
+      Given path 'inventory/items'
+      And request
+      """
+      {
+        "status":{"name":"Available"},
+        "holdingsRecordId":"#(holdingsId)",
+        "materialType":{"id":"d9acad2f-2aac-4b48-9097-e6ab85906b25"},
+        "permanentLoanType":{"id":"2e48e713-17f3-4c13-a9f8-23845bb210a4"}
+      }
+      """
+      When method POST
+      Then status 201
+      * def dupItemHrId = response.id
+      * def dupItemHrId = response.hrid
+#     Both Items id's & hrId's should not match
+      * match dupItemHrId != itemId
+      * match dupItemHrId != itemHrId
+
+#     Deletion check
+#     If an item is associated with Holdings then holdings can't be deleted.
+      * def expected_response = 'Cannot delete holdings_record.id = ' + holdingsId + ' because id is still referenced from table item.'
+      Given path '/holdings-storage/holdings/' + holdingsId
+      And request
+      """
+      {
+        "status":{"name":"Available"},
+        "holdingsRecordId":"#(holdingsId)",
+        "materialType":{"id":"d9acad2f-2aac-4b48-9097-e6ab85906b25"},
+        "permanentLoanType":{"id":"2e48e713-17f3-4c13-a9f8-23845bb210a4"}
+      }
+      """
+      When method DELETE
+      Then status 400
+      * match expected_response == response
```

Comment:
I suggest we follow the [structuring advice](https://wiki.folio.org/display/FOLIJET/Karate+setup) linked from the [official wiki page](https://wiki.folio.org/display/FOLIJET/Karate+API+Integration+Testing) on these tests.

That seems to suggest we have an overall feature for set up and separate JUnit tests and feature files for specific areas of behaviour.

Reply:
@marcjohnson-kint

Reply:
I agree with your suggestion , but in case of MOD-INVENTORY tests we don't have distinct behavior . All tests are related to inventory module only. In case of MOD-INVENTORY karate tests,  we are not  integrating with any other module. 
For example , in case of **Mod-Circulation** we can create multiple feature files because it integrates with other modules also . so for each module we can create a separate feature file.

Reply:
> I agree with your suggestion , but in case of MOD-INVENTORY tests we don't have distinct behavior . All tests are related to inventory module only.

How is that different to the example in the documentation that is for mod-orders?

I think we do have groups of behaviour.

I think there are a few examples of groupings in the existing tests we could adopt:
* creation of the hierarchy (instance, holdings, item)
* moving a holdings or item
* derivation of effective location (could be further divided into holdings and item levels)
* preceding and succeeding titles

bq. In case of MOD-INVENTORY karate tests, we are not integrating with any other module. For example , in case of **Mod-Circulation** we can create multiple feature files because it integrates with other modules also . so for each module we can create a separate feature file.

I think you are suggesting we want to organise the scenarios by what other modules are involved.

How wrong is my interpretation of this statement?

Why would we organise features by integration?

Reply:
Lets take an example of a simple MOD-CIRCULATION test case
1:-Create Karate test for Loans: When patron and item id's entered at checkout, post a new loan using the circulation rule matched (circ-3)

So in this case we first need to have 
1:- An Instance 
2:-servicepoint
3:-Location
4:-Holdings
5:-Item
6:-loan policy
7:-User and group
These things are coming from Mod-Inventory, Mod-Users etc.
so we can create a separate feature file for each module. 


But in **mod-inventory** & **mod-calendar** i dont see  any test case required data from other modules, thats why i think creating seperate feature file is not a good idea. let me know for any suggestion

Reply:
Lets take an example of a simple MOD-CIRCULATION test case
1:-Create Karate test for Loans: When patron and item id's entered at checkout, post a new loan using the circulation rule matched (circ-3)

So in this case we first need to have
1:- An Instance
2:-servicepoint
3:-Location
4:-Holdings
5:-Item
6:-loan policy
7:-User and group
These things are coming from Mod-Inventory, Mod-Users etc.
so we can create a separate feature file for each module.

But in mod-inventory & mod-calendar i dont see any test case required data from other modules, thats why i think creating seperate feature file is not a good idea. let me know if any suggestions.

Reply:
> in mod-inventory & mod-calendar i dont see any test case required data from other modules, thats why i think creating seperate feature file is not a good idea. let me know if any suggestions.

I've suggested we decompose features by areas of behaviour rather than integration.

I get the sense that you either disagree with this approach or don't think it is possible / relevant for inventory (as you considered all of the examples I gave above as the same group).

I'm not sure how to continue the conversation.

Reply:
Yes , i dont think **decomposing features by areas of behaviour** is relevant for inventory.

Reply:
Then I think we are at an impasse and there is no where for the conversation to go.